File: 0001-two-sum/0001-two-sum.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mpp;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            int complement = target - nums[i];
            if (mpp[complement] != 0) {
                return {mpp[complement]-1, i}; 
            }
            mpp[nums[i]] = i+1;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val mp = HashMap<Int,Int>()
        var i = 1
        for(x in nums){
            if((target - x) in mp)return intArrayOf(mp[target-x]!!-1,i-1)
            mp[x] = i
            i++
        }
        return intArrayOf()
    }
}